src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900104.s")
+    if (this->unk_311 == 0) {
+        if (((this->unk_198 - 4.0f) < this->skelAnime.curFrame) &&
+            (this->skelAnime.curFrame < (this->unk_198 + 6.0f))) {
+            this->unk_312 = 1;
+        }
+    } else if ((((this->unk_198 - 4.0f) + 4.0f) < this->skelAnime.curFrame) &&
+               (this->skelAnime.curFrame < (this->unk_198 + 6.0f))) {
+        this->unk_312 = 2;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900210.s")
+    Math_ApproachZeroF(&this->actor.speedXZ, 0.5f, 1.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090026C.s")
+    if (Animation_OnFrame(&this->skelAnime, this->unk_194)) {
+        this->unk_311 = 1 - this->unk_311;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_809002CC.s")
+        if ((this->unk_311 == 1) && (this->actor.xzDistToPlayer < 250.0f) && (this->unk_313 != 0)) {
+            func_80900580(this, globalCtx);
+        } else {
+            func_808FFDB0(this, globalCtx);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900344.s")
+    func_808FFAC8(this, globalCtx, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900580.s")
+    if ((this->unk_334 == 0) && (this->unk_311 == 0)) {
+        this->unk_320 = 0.0f;
+    } else {
+        func_808FFBBC(this, globalCtx, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900650.s")
+void func_80900818(BossGanon2* this, GlobalContext* globalCtx) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &object_ganon2_Anim_02A848, -5.0f);
+    this->unk_194 = Animation_GetLastFrame(&object_ganon2_Anim_02A848);
+    this->actionFunc = func_80900890;
+    this->unk_1AC = 0;
+    this->unk_39C = 0;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MGANON_DEAD1);
+    this->unk_336 = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900818.s")
+void func_80900890(BossGanon2* this, GlobalContext* globalCtx) {
+    Vec3f sp5C;
+    Vec3f sp50;
+    Camera* sp4C;
+    Player* player;
+    Camera* temp_v0;
+    Camera* temp_v0_2;
+    s32 pad;
+    f32 temp_f12;
+    f32 temp_f2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900890.s")
+    sp4C = Gameplay_GetCamera(globalCtx, 0);
+    player = PLAYER;
+    SkelAnime_Update(&this->skelAnime);
+    this->unk_398++;
+    this->unk_339 = 0x14;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80901020.s")
+    switch (this->unk_39C) {
+        case 0:
+            func_80064520(globalCtx, &globalCtx->csCtx);
+            this->unk_39E = Gameplay_CreateSubCamera(globalCtx);
+            Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+            Gameplay_ChangeCameraStatus(globalCtx, this->unk_39E, 7);
```

Comment:
```suggestion
            Gameplay_ChangeCameraStatus(globalCtx, MAIN_CAM, CAM_STAT_WAIT);
            Gameplay_ChangeCameraStatus(globalCtx, this->unk_39E, CAM_STAT_ACTIVE);
```
same in other places